chia/wallet/wallet_state_manager.py

```diff
@@ -596,7 +596,9 @@ def tx_pending_changed(self) -> None:
 
         self.pending_tx_callback()
 
-    async def synced(self) -> bool:
+    async def synced(self, block_is_current_at: Optional[int] = None) -> bool:
```

Comment:
I can't find a place in this patch where `block_is_current_at` is passed in, when calling this function. Without a lot of context; it seems totally reasonable to be able to avoid have this function ask for what time it is. But it would seem simpler to just pass in the current time, rather than the time at which we consider blocks to be synced (i.e. 7 minutes ago, from the current time).

This same comment applies to `full_node.py` as well.

You could just take `now` as an optional parameter.

Reply:
> I can't find a place in this patch where block_is_current_at is passed in, when calling this function

Yeah the patch adds the _ability_ to override the "When is the block current" behavior if desired.  Then in `conftest.py` we monkeypatch in a new default of `0` making all blocks current.  The implicit constants of 5 and 7 minutes were causing issues so this gives us the ability to patch those numbers.

> You could just take now as an optional parameter.

As stated above, the issue is not the "now" so much as the "since when".  This value is pretty handy if you view the function like `.am_i_syned_since(this_time):`.